cli/src/pcluster/schemas/cluster_schema.py

```diff
@@ -45,80 +109,302 @@ def evaluate_dynamic_defaults(self, raw_data, **kwargs):
     def make_config(self, data, **kwargs):
         """Generate config object."""
-        return ImageConfig(**data)
+        return BaseEbsConfig(**data)
+
 
+class RaidSchema(BaseSchema):
+    """Represent the schema of the parameters specific to Raid. It is a child of EBS schema."""
 
-class HeadNodeNetworkingSchema(BaseSchema):
+    raid_type = fields.Str(data_key="Type", validate=validate.OneOf(["0", "1"]))
+    number_of_volumes = fields.Int(validate=validate.Range(min=2, max=5))
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return RaidConfig(**data)
+
+
+class EbsSchema(BaseEbsSchema):
+    """Represent the schema of EBS."""
+
+    snapshot_id = fields.Str(validate=validate.Regexp(r"^snap-[0-9a-z]{8}$|^snap-[0-9a-z]{17}$"))
+    volume_id = fields.Str(validate=validate.Regexp(r"^vol-[0-9a-z]{8}$|^vol-[0-9a-z]{17}$"))
+    raid = fields.Nested(RaidSchema)
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return EbsConfig(**data)
+
+
+class EphemeralVolumeSchema(BaseSchema):
+    """Represent the schema of ephemeral volume.It is a child of storage schema."""
+
+    encrypted = fields.Bool()
+    mount_dir = fields.Str(validate=_get_field_validator("file_path"))
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return EphemeralVolumeConfig(**data)
+
+
+class StorageSchema(BaseSchema):
+    """Represent the schema of root volume."""
+
+    root_volume = fields.Nested(BaseEbsSchema)
+    ephemeral_volume = fields.Nested(EphemeralVolumeSchema)
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return StorageConfig(**data)
+
+
+class EfsSchema(BaseSchema):
+    """Represent the schema of EFS."""
+
+    encrypted = fields.Bool()
+    kms_key_id = fields.Str()
+    performance_mode = fields.Str(validate=validate.OneOf(["generalPurpose", "maxIO"]))
+    throughput_mode = fields.Str(validate=validate.OneOf(["provisioned", "bursting"]))
+    provisioned_throughput = fields.Int(validate=validate.Range(min=0, max=1024))
+    file_system_id = fields.Str(validate=validate.Regexp(r"^fs-[0-9a-z]{8}$|^fs-[0-9a-z]{17}$"))
+
+    @post_load
+    def make_config(self, data, **kwargs):
+        """Generate config object."""
+        return EfsConfig(**data)
+
+    @validates_schema
+    def validate_existence_of_mode_throughput(self, data, **kwargs):
+        """Validate the conditional existence requirement between throughput_mode and provisioned_throughput."""
+        throughput_mode = data.get("throughput_mode")
+        provisioned_throughput = data.get("provisioned_throughput")
+        if throughput_mode != "provisioned" and provisioned_throughput:
+            raise ValidationError(
+                message="When specifying 'provisioned_throughput', the 'throughput_mode' must be set to 'provisioned'",
```

Comment:
the message should be cleaned up to talk about provisioned througput and not "provisioned_througput" with the underscore in the middle that is the name of the old configuration parameter.

Reply:
I changed this one. There are some remaining messages to be cleaned up. I will change them once I have time.